mpt_trie/src/nibbles.rs

```diff
@@ -74,11 +91,59 @@ pub enum FromHexPrefixError {
     /// The hex prefix encoding flag is invalid.
     InvalidFlags(Nibble),
 
-    #[error("Tried to convert a hex prefix byte string into `Nibbles` that was longer than 64 bytes: (length: {0}, bytes: {1})")]
+    #[error("Tried to convert a hex prefix byte string into `Nibbles` that was longer than 40 bytes: (length: {0}, bytes: {1})")]
     /// The hex prefix encoding is too large.
     TooLong(String, usize),
 }
 
+/// Error type for conversion.
+#[derive(Debug, Error)]
+pub enum NibblesToTypeError {
+    #[error("Overflow encountered when converting to U256: {0}")]
+    /// Overflow encountered.
+    Overflow(NibblesIntern),
+}
+
+trait AsU64s {
+    fn as_u64s(&self) -> Vec<u64>;
+}
+
+macro_rules! impl_as_u64s_for_primitive {
+    ($type:ty) => {
+        impl AsU64s for $type {
+            fn as_u64s(&self) -> Vec<u64> {
+                vec![*self as u64]
+            }
+        }
+    };
+}
```

Comment:
Just a random thought as someone who has not had to put time into getting this PR to work. Would it make sense for this trait to instead return an `impl Iterator<Item = u64>`? Then we can avoid the `Vec` allocation in the `impl_to_nibbles` impls below.